src/main/java/seedu/address/model/task/Task.java

```diff
@@ -0,0 +1,151 @@
+package seedu.address.model.task;
+
+import java.util.Objects;
+
+/**
+ * The Task class represents a general task with a description and a completion status.
+ * It serves as the base class for more specific types of tasks such as Todo, Deadline, and Event.
+ */
+public class Task {
+    protected String description;
+    protected boolean isDone;
+
+    /**
+     * Constructs a Task with the specified description.
+     * The task is initially marked as not done.
+     *
+     * @param description The description of the task.
+     */
+    public Task(String description) {
+        this.description = description;
+        this.isDone = false;
+    }
+
+    /**
+     * Returns the status icon of the task.
+     * "X" if the task is done, " " if the task is not done.
+     *
+     * @return The status icon of the task.
+     */
+    public String getStatusIcon() {
+        return (isDone ? "X" : " "); // mark done task with X
+    }
+
+    /**
+     * Returns the description of the task.
+     *
+     * @return The description of the task as a String.
+     */
+    public String getDescription() {
+        return this.description;
+    }
+
+    /**
+     * Returns the isDone status the task.
+     *
+     * @return The completion status of the task as a Boolean.
+     */
+    public Boolean getIsDone() {
+        return this.isDone;
+    }
+
+    /**
+     * Marks the task as done.
+     */
+    public void markAsDone() {
+        isDone = true;
+    }
+
+    /**
+     * Marks the task as not done.
+     */
+    public void markAsUndone() {
+        isDone = false;
+    }
+
+    /**
+     * Returns the presence of the given keyword within the description.
+     * "true" if the keyword is present, "false" if the keyword is not.
+     *
+     * @param keyword The supplied keyword string.
+     * @return The boolean indicating presence of the keyword
+     */
+    public boolean hasKeyword(String keyword) {
+        String[] descriptionArray = description.split(" ");
+        for (String word : descriptionArray) {
+            if (word.equals(keyword)) {
+                return true;
+            }
+        }
+        return false;
+    }
+
+    /**
+     * Returns the presence of the given keyword if it is
+     * within a partial/full word in the description, case-insensitive.
+     * "true" if the keyword is present, "false" if the keyword is not.
+     *
+     * @param keyword The supplied keyword string.
+     * @return The boolean indicating presence of the keyword
+     */
+    public boolean hasKeywordInPartialDescription(String keyword) {
```

Comment:
True, this will be able to do what hasKeyword does, I think I will just remove the hasKeyword function since its quite limited and we likely won't be using it.